packages/samples/headless-ssr-commerce/components/parameter-manager.tsx

```diff
@@ -0,0 +1,52 @@
+'use client';
+
+import {useParameterManager} from '@/lib/commerce-engine';
+import {useAppHistoryRouter} from '@coveo/headless-react/ssr-commerce';
+import {buildSSRCommerceSearchParameterSerializer} from '@coveo/headless/ssr-commerce';
+import {useEffect, useMemo} from 'react';
+
+export default function ParameterManager() {
+  const historyRouter = useAppHistoryRouter();
+
+  const {state, controller} = useParameterManager();
+
+  useEffect(() => {
+    if (!controller || !historyRouter.url?.searchParams) {
+      return;
+    }
+    const {toCommerceSearchParameters} =
+      buildSSRCommerceSearchParameterSerializer();
+    const searchParameters = toCommerceSearchParameters(
+      historyRouter.url.searchParams
+    );
+    return controller.synchronize(searchParameters);
+  }, [historyRouter.url?.searchParams, controller]);
+
+  const correctedUrl = useMemo(() => {
+    if (!historyRouter.url) {
+      return null;
+    }
+    const newURL = new URL(historyRouter.url);
+    const {serialize} = buildSSRCommerceSearchParameterSerializer();
+
+    return serialize(state.parameters, newURL);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Which deps are we missing, and why :suspect: ?

Reply:
This is the fruit of copy-pasting

I added the missing deps and it's all working fine.

Reply:
Upon further investigation, there's a reason for this: including the deps prevents navigation with the browser's back/forward buttons. I'm not entirely sure why. @y-lakhdar maybe you have some insight into this?

Reply:
For this one, `useMemo` recomputes the memoized value when one of the deps has changed, so I think that would explain it.